### DISCUSSION TITLE
Question about multithreading

### DISCUSSION THREAD

**Discussion by duanqn** [zxing-cpp/zxing-cpp#395] [Q&A] [answered]
Hi,
I would like to get some confirmation on threading-related stuff.
As of the current HEAD version, when I call `ZXing::ReadBarcodes`, is it guaranteed to use a single thread?
If I call `ZXing::ReadBarcodes` in multiple threads, is it safe to do so or do I need additional locks?

In other words, if I have a lot of images for barcode recognition on a 4 core machine, should I start 4 threads and call `ZXing::ReadBarcodes` in parallel?

**Comment by axxel** [comment, marked as answer]
1) Yes the current HEAD revision does not spawn threads on it's own.
2) Yes, you can run multiple threads and execute `ReadBarcodes` in parallel. (Note: that is not currently tested properly, if an issue arises, please let us know.)

**Comment by axxel** [comment]
Did you gain some experience with multi-threaded `ReadBarcodes` execution?

**Comment by duanqn** [comment]
> Did you gain some experience with multi-threaded `ReadBarcodes` execution?

Not yet...

**Comment by duanqn** [comment]
Ran some tests and everything worked fine.
